### ISSUE TITLE
ldap outpost panics and crashes when doing an LDAP search with a %20 in the search filter

### ISSUE THREAD

**Issue by a-gerhard** [goauthentik/authentik#4662]
**Describe the bug**
I had a small misconfiguration in my LDAP client that resulted in a `%20` appearing in the ldap search filter sent to the LDAP outpost. This caused the outpost program to panic. This doesn't seem to affect all searches with a % in it though. The same search works fine when having an actual space in the filter instead.

**To Reproduce**
1. Set up an LDAP outpost and create a user that can normally run ldapsearch
2. run an LDAP search with a search filter like `(&(objectClass=user)(memberOf=cn=authentik%20Admins,ou=groups,dc=redacted))`

**Expected behavior**
LDAP search should try to match this, or return an Error (I am unsure if % is specified in DNs), but the program shouldn't crash.

**Logs**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe42805]
goroutine 85 [running]:
goauthentik.io/internal/outpost/ldap/search/direct.(*DirectSearcher).Search.func1()
	/go/src/goauthentik.io/internal/outpost/ldap/search/direct/direct.go:135 +0x6e5
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.0.0-20220601150217-0de741cfad7f/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/pkg/mod/golang.org/x/sync@v0.0.0-20220601150217-0de741cfad7f/errgroup/errgroup.go:72 +0xa5
```

**Version and Deployment (please complete the following information):**
 - authentik version: 2023.1.2
 - Deployment: docker-compose

**Comment by BeryJu** [comment]
This should already be fixed in main, can you try with the beta? https://goauthentik.io/docs/installation/beta

**Comment by a-gerhard** [comment]
I'd rather not deploy a beta version in production.

I'll let you know when I have upgraded to the next release version.

**Comment by t3chguy** [comment]
I was seeing the issue and updating to beta cleared it

**Comment by a-gerhard** [comment]
Issue has been resolved with the 2023.2.2 update
